### DISCUSSION TITLE
Distinguish between forked PR and own PR

### DISCUSSION THREAD

**Discussion by cytopia** [community/community#25820] [Actions]
<p>I want to setup a workflow that only runs on PR’s (so that my PR’s are triggered and also PR’s from people who forked my repository).</p>
<p>The workflow does:</p>
<ol>
<li>Build docker image</li>
<li>Test docker image</li>
<li>Pushes docker image to dockerhub</li>
</ol>
<p>I want to make sure, that step **3.** only happens on the PR’s I create myself, but never on PR’s that come from forks.</p>
<p>Is there a way to figure out (e.g.: via <code>env</code> or similar) if the current job runs on a PR that was initially forked or from my own repository?</p>

**Comment by dawidd6** [comment]
<p>Yea, you won’t be able to use secrets in Actions run from forks. So if you use secrets to publish this image, it won’t succeed for forks anyway.</p>

**Comment by cytopia** [comment]
<p>I want to explicitly NOT run any steps that require secrets on forked PR’s via some kind of condition.<br>
Everything that comes from myself should allow this step to run.</p>

**Comment by dawidd6** [comment]
<pre><code>steps:
  - name: Publish image (only repo owner)
    if: github.event.repository.owner.id == github.event.pull_request.user.id
</code></pre>
<p>You can try something like this.</p>
<p>Reference: <a href="https://developer.github.com/v3/activity/events/types/#pullrequestevent" rel="nofollow noopener">https://developer.github.com/v3/activity/events/types/#pullrequestevent</a></p>

**Comment by cytopia** [comment]
<p>I tried exactly this, but it doesn’t work.</p>
<p>I’ve defined it here as suggested:</p>
<p><a href="https://github.com/cytopia/docker-ansible/blob/release-0.16-1/.github/workflows/linting.yml#L39" rel="nofollow noopener">https://github.com/cytopia/docker-ansible/blob/release-0.16-1/.github/workflows/linting.yml#L39</a></p>
<p>Then I’ve created a pull request from a forker account to verify this. The build step runs regardless:<br>
<a href="https://github.com/cytopia/docker-ansible/pull/33/checks?check_run_id=277956369#step:4:1" rel="nofollow noopener">https://github.com/cytopia/docker-ansible/pull/33/checks?check_run_id=277956369#step:4:1</a></p>
<p>And </p>

**Comment by dawidd6** [comment, marked as answer]
<pre><code>if: github.event.pull_request.base.repo.id == github.event.pull_request.head.repo.id
</code></pre>
<p>I misunderstood you a bit, try this one.</p>

**Comment by cytopia** [comment]
<p>Thanks, this works as expected</p>
